app/controllers/analytics/districts_controller.rb

```diff
@@ -0,0 +1,47 @@
+class Analytics::DistrictsController < AnalyticsController
+  before_action :set_district
+  before_action :set_organization
+  before_action :set_facilities
+
+  def show
+    @days_previous = 20
+    @months_previous = 8
+
+    @district_analytics = district_analytics(@from_time, @to_time)
+    @facility_analytics = facility_analytics(@from_time, @to_time)
+  end
+
+  def graphics
+    @current_month = Date.today.at_beginning_of_month.to_date
+    @district_analytics = @district.patient_set_analytics(@from_time, @to_time)
+  end
+
+  private
+
+  def set_district
+    district_id = params[:id] || params[:district_id]
+    @district = District.new(district_id)
+    authorize(@district)
+  end
+
+  def set_organization
+    @district.organization_id = params[:organization_id]
```

Comment:
We can just make this an optional constructor argument.

Reply:
We can instead set `@organization` in the controller.

Reply:
Updated to inject `@organization`, got rid of references to `organization_id`, fixed failing specs.